### PR TITLE
[feature] Rotator can write CircleCI env variables

### PR DESCRIPTION
**Pull request by edulop91**

### CODE REVIEW

**Comment by mahmoudian1** [comment]
# [Codecov](https://codecov.io/gh/chanzuckerberg/rotator/pull/41?src=pr&el=h1) Report
> Merging [#41](https://codecov.io/gh/chanzuckerberg/rotator/pull/41?src=pr&el=desc) into [master](https://codecov.io/gh/chanzuckerberg/rotator/commit/54f75827cebb4a288741b82d4cf34792ba524c9b&el=desc) will **increase** coverage by `0.57%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/chanzuckerberg/rotator/pull/41?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master      #41      +/-   ##
==========================================
+ Coverage   27.27%   27.84%   +0.57%     
==========================================
  Files          12       13       +1     
  Lines         385      413      +28     
==========================================
+ Hits          105      115      +10     
- Misses        269      287      +18     
  Partials       11       11              
```


| [Impacted Files](https://codecov.io/gh/chanzuckerberg/rotator/pull/41?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [pkg/sink/circleCi.go](https://codecov.io/gh/chanzuckerberg/rotator/pull/41/diff?src=pr&el=tree#diff-cGtnL3NpbmsvY2lyY2xlQ2kuZ28=) | `83.33% <0.00%> (ø)` | |

------

[Continue to review full report at Codecov](https://codecov.io/gh/chanzuckerberg/rotator/pull/41?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/chanzuckerberg/rotator/pull/41?src=pr&el=footer). Last update [54f7582...f36e618](https://codecov.io/gh/chanzuckerberg/rotator/pull/41?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: README.md [hunk 1]
```diff
@@ -112,6 +112,14 @@ All sinks must have the following fields in addition to any sink-specific fields
 
 [`TRAVIS_API_AUTH_TOKEN`](https://github.com/shuheiktgw/go-travis#authentication-with-travis-api-token) should be set.
 
+### Circle CI (`CircleCI`)
+| Name | Description | Required |
+|------|-------------|:-----:|
+| account | The CircleCI account to write this env var to. | yes |
+| repo | The CircleCI repo to write this env var to. | yes |
+
+[`CIRCLECI_AUTH_TOKEN`](https://circleci.com/docs/2.0/managing-api-tokens/) should be set.
```
Comment (ryanking) [inline README.md]:
```suggestion
[`CIRCLECI_AUTH_TOKEN`](https://circleci.com/docs/2.0/managing-api-tokens/) must be set.
```

File: pkg/config/config.go [hunk 2]
```diff
@@ -161,8 +162,17 @@ func unmarshalSinks(sinksIface interface{}) (sink.Sinks, error) {
...
 				return nil, errors.Wrap(err, "unable to authenticate travis API")
 			}
-
 			sinks = append(sinks, &sink.TravisCiSink{BaseSink: sink.BaseSink{KeyToName: keyToName}, RepoSlug: sinkMapStr["repo_slug"], Client: client})
+
+		case sink.KindCircleCi:
+			if err = validate(sinkMapStr, "account", "repo"); err != nil {
+				return nil, errors.Wrap(err, "missing keys in circle CI sink config")
+			}
+			client := &circleci.Client{Token: os.Getenv("CIRCLECI_AUTH_TOKEN")}
```
Comment (ryanking) [inline pkg/config/config.go]: Correct me if I am wrong, but there is no way for writing to circle to work without this env var, right?

If that's the case we should use [LookupEnv](https://golang.org/src/os/env.go?s=3235:3276#L102) and return a good error if the env variable is not set.

**Comment by mahmoudian1** [comment]
# [Codecov](https://codecov.io/gh/chanzuckerberg/rotator/pull/41?src=pr&el=h1) Report
> Merging [#41](https://codecov.io/gh/chanzuckerberg/rotator/pull/41?src=pr&el=desc) into [master](https://codecov.io/gh/chanzuckerberg/rotator/commit/54f75827cebb4a288741b82d4cf34792ba524c9b&el=desc) will **increase** coverage by `0.23%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/chanzuckerberg/rotator/pull/41?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master      #41      +/-   ##
==========================================
+ Coverage   27.27%   27.51%   +0.23%     
==========================================
  Files          12       13       +1     
  Lines         385      418      +33     
==========================================
+ Hits          105      115      +10     
- Misses        269      292      +23     
  Partials       11       11              
```


| [Impacted Files](https://codecov.io/gh/chanzuckerberg/rotator/pull/41?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [pkg/sink/circleCi.go](https://codecov.io/gh/chanzuckerberg/rotator/pull/41/diff?src=pr&el=tree#diff-cGtnL3NpbmsvY2lyY2xlQ2kuZ28=) | `83.33% <0.00%> (ø)` | |

------

[Continue to review full report at Codecov](https://codecov.io/gh/chanzuckerberg/rotator/pull/41?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/chanzuckerberg/rotator/pull/41?src=pr&el=footer). Last update [54f7582...a4f184f](https://codecov.io/gh/chanzuckerberg/rotator/pull/41?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Pull request merged**